### PR TITLE
React router configuration

### PR DESCRIPTION
**Pull request by ductiletoaster**
Supporting nested routes and 404s

### CODE REVIEW

File: src/shared/App.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
 import * as React from "react";
 import {Switch} from "react-router-dom";
-import {SingleRoute} from "./components/atoms";
-import {NoMatch} from "./components/pages";
+import {Route} from "./components/atoms";
```
Comment (jimmynotjim) [inline src/shared/App.tsx]: I know this file is in `shared`, but should these be updated to use `@shared/atoms` that way if it's ever moved it continues to work? If yes, we should update all of them.

Comment (ductiletoaster) [reply src/shared/App.tsx]: Yeah, I'm down to update them. Just didn't want to update all at once. We can follow up in another PR?

File: src/shared/components/templates/Default/Default.tsx [hunk 2]
```diff
@@ -1,17 +1,14 @@
+import {Route, RouteComponentProps} from "@shared/components";
 import * as React from "react";
-import { Link, RouteComponentProps } from "react-router-dom";
+import {Link, Switch} from "react-router-dom";
```
Comment (jimmynotjim) [inline src/shared/components/templates/Default/Default.tsx]: We should add spaces inside `{}` to the linter, it's a bit inconsistent with some having a space and some not. No preference on which we go with but just so we know what it is.

**Comment by jimmynotjim** [comment]
Sorry I didn't finish reviewing till this morning. No blocking issues and everything can be addressed in the future.

**Pull request merged**